security/pkg/server/ca/server.go

```diff
@@ -168,14 +164,10 @@ func recordCertsExpiry(keyCertBundle *util.KeyCertBundle) {
 	certChainExpiry, err := keyCertBundle.ExtractCACertExpiryTimestamp()
 	if err != nil {
 		serverCaLog.Errorf("failed to extract CA cert expiry timestamp (error %v)", err)
+	} else {
+		certChainExpiryTimestamp.Record(float64(certChainExpiry.Unix()))
+		certChainExpirySeconds.ValueFrom(time.Until(*certChainExpiry).Seconds)
```

Comment:
certChainExpirySeconds and certChainExpiryTimestamp are much like. Probably one metric is enough

Reply:
I agree, but this was created to get around a datadog limitation, and getting rid of this metric after adding it seems wrong. #51037

Reply:
Yes, we should not remove or change existing metrics. My concern was making sure the metric ( as badly named as it is - like plenty of other badly named things ) - is usable in all modes and correct ( i.e. not only the first cert )

Reply:
BTW - even for cert chain - it's complicated :-) 

Like the Let's encrypt change - you can have a chain that has cross-signing and multiple roots - the alert should be triggered if the chain is about to become invalid, but if a path to a non-expiring root exists - no need to alert. 

For example I may have a root that is expiring - and start issuing intermediaries signed with both the old root and new root. That's valid and works with clients that still have the old root and not the new root - and not a reason to alert, since it works as intended even if some certs in the chain are about to expire.